src/async/pitfalls/cancellation.md

```diff
@@ -0,0 +1,91 @@
+# Cancellation
+
+Dropping a future implies it can never be polled again. This is called *cancellation*.
+It can occur at any `await` point. Care is needed with stateful resources.
```

Comment:
When I talked about this in the class, I compared it to panic -- Rust will free any resources by calling destructors, deallocating, etc., but it may still be an unexpected control flow.  But in most cases we think of panic as killing the process, whereas it's much more common for a program to recover from an async cancellation.

I'm not sure that's a helpful way to frame it, but there it is :)

Reply:
Cancellation-safety is now defined in the main slide, but not given as a term... I added some speaker notes about the compiler, and comparing cancellation to other control flow.